xbmc/pvr/epg/EpgInfoTag.cpp

```diff
@@ -82,19 +103,21 @@ CPVREpgInfoTag::CPVREpgInfoTag(const EPG_TAG &data, int iClientId)
   UpdatePath();
 }
 
+void CPVREpgInfoTag::SetChannelData(const std::shared_ptr<CPVREpgChannelData>& data)
+{
+  CSingleLock lock(m_critSection);
```

Comment:
Why do you lock the critical section here for `m_channelData` and almost nowhere else?

Reply:
For no reason. m_channeldata access must always be mutex protected. Thanks.